### DISCUSSION TITLE
C++ library and examples for custom (desktop) player

### DISCUSSION THREAD

**Discussion by cemakaus** [OvenMediaLabs/OvenMediaEngine#1197] [Help]
Hello,

I have a task to collect video captures from cameras and distribute those streams to several processing software. I want to put OME between capture software instances and consumers (processing software instances) for multiplexing. I've chosen webrtc for low latency. I'm developing in C++ and new to webrtc and OME. Could you please point libraries or frameworks that can be used to develop such software (for the transmitting and receiving video) and some examples to follow?

(Additionally: Is it possible to use gstreamer with OME, if so, how?)

**Comment by getroot** [comment]
OME supports WHIP, so use WHIP when sending streams from your encoder to OME. This will allow your encoder to connect to OME as well as other servers that support WHIP. You may find it helpful to see a recent thread supporting WHIP in OBS. In the discussion below, you will also find the process of selecting WebRTC libraries such as webrtc native code and libdatachannel in OBS.

https://github.com/obsproject/obs-studio/pull/7926

For playback, use the WebRTC library of your choice, and for signaling follow OME's specifications.

https://airensoft.gitbook.io/ovenmediaengine/streaming/webrtc-publishing#signalling-protocol

For signaling example code, see OvenPlayer (https://github.com/AirenSoft/OvenPlayer) or OvenRtcTester (https://github.com/AirenSoft/OvenMediaEngine/tree/master/misc/oven_rtc_tester).
